homeassistant/components/cert_expiry/sensor.py

```diff
@@ -15,62 +13,93 @@
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util import dt as dt_util
 
-from .const import DEFAULT_NAME, DEFAULT_PORT, DOMAIN
-from .helper import get_cert
+from .const import DEFAULT_PORT, DOMAIN
+from .errors import TemporaryFailure, ValidationFailure
+from .helper import get_cert_time_to_expiry
 
 _LOGGER = logging.getLogger(__name__)
 
 SCAN_INTERVAL = timedelta(hours=12)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    }
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_NAME, invalidation_version="0.109"),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        }
+    ),
 )
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up certificate expiry sensor."""
 
+    @callback
+    def schedule_import(_):
+        """Schedule delayed import after HA is fully started."""
+        import_time = dt_util.utcnow() + timedelta(seconds=10)
+        async_track_point_in_utc_time(hass, do_import, import_time)
+
     @callback
     def do_import(_):
-        """Process YAML import after HA is fully started."""
+        """Process YAML import."""
         hass.async_create_task(
             hass.config_entries.flow.async_init(
                 DOMAIN, context={"source": SOURCE_IMPORT}, data=dict(config)
             )
         )
 
-    # Delay to avoid validation during setup in case we're checking our own cert.
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, do_import)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, schedule_import)
 
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Add cert-expiry entry."""
+    days = 0
+    error = None
+    hostname = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+
+    if entry.unique_id is None:
+        hass.config_entries.async_update_entry(entry, unique_id=f"{hostname}:{port}")
+
+    try:
+        days = await get_cert_time_to_expiry(hass, hostname, port)
+    except TemporaryFailure as err:
+        _LOGGER.error(err)
+        raise PlatformNotReady
+    except ValidationFailure as err:
+        error = err
+
     async_add_entities(
-        [SSLCertificate(entry.title, entry.data[CONF_HOST], entry.data[CONF_PORT])],
-        False,
-        # Don't update in case we're checking our own cert.
+        [SSLCertificate(hass, hostname, port, days, error)], False,
     )
     return True
 
 
 class SSLCertificate(Entity):
     """Implementation of the certificate expiry sensor."""
 
-    def __init__(self, sensor_name, server_name, server_port):
+    def __init__(self, hass, server_name, server_port, days, error):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
I think I added this in before the config flow tests were using the core calls. Works fine now without setting it manually, thanks.